test/integration/authorizeRoles.test.js

```diff
@@ -0,0 +1,197 @@
+const chai = require("chai");
+const { expect } = chai;
+
+const { authorizeRoles } = require("../../middlewares/authorizeRoles");
+const authenticate = require("../../middlewares/authenticate");
+const authService = require("../../services/authService");
+const addUser = require("../utils/addUser");
+const cleanDb = require("../utils/cleanDb");
+const config = require("config");
+const cookieName = config.get("userToken.cookieName");
+const userData = require("../fixtures/user/user")();
+
+const defaultUser = userData[0]; // user with no `roles` key
+const appOwner = userData[3];
+const superUser = userData[4];
+
+// Route setup with various permissions for testing
+const express = require("express");
+const router = express.Router();
+const AppMiddlewares = require("../../middlewares");
+
+const pongHandler = (_, res) => {
+  return res.json({ message: "pong" });
+};
+
+router.get("/for-app-owner", authenticate, authorizeRoles(["app_owner"]), pongHandler);
+router.get("/for-super-user", authenticate, authorizeRoles(["super_user"]), pongHandler);
+router.get("/for-super-user-and-app-owner", authenticate, authorizeRoles(["super_user", "app_owner"]), pongHandler);
+router.get("/for-invalid", authenticate, authorizeRoles(["invalid"]), pongHandler);
+
+const app = express();
+AppMiddlewares(app);
+app.use("/", router);
+
+describe("authorizeRoles", function () {
+  let defaultJwt, appOwnerJwt, superUserJwt;
+
+  before(async function () {
+    const defaultUserId = await addUser(defaultUser);
+    const appOwnerId = await addUser(appOwner);
+    const superUserId = await addUser(superUser);
+    defaultJwt = authService.generateAuthToken({ userId: defaultUserId });
+    appOwnerJwt = authService.generateAuthToken({ userId: appOwnerId });
+    superUserJwt = authService.generateAuthToken({ userId: superUserId });
+  });
+
+  after(async function () {
+    await cleanDb();
+  });
```

Comment:
Can we rename these two to beforeEach and afterEach. Increases Readibility.

Reply:
Those are different functions.